### PR TITLE
Interconnect v 800

### PR DESCRIPTION
**Pull request by jyothis-gireesan**
**Description**
API 800 Support for Interconnect 

**Check List**

- [X] New functionality includes testing.
- [ ]  All tests pass for go 1.11 + gofmt checks .
- [X] New functionality has been documented in the README if applicable.
- [X] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [X] New endpoints supported are updated in the endpoints-support.md file.
- [ ]  Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/interconnect.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	)
+	ovc := clientOV.NewOVClient(
+		os.Getenv("ONEVIEW_OV_USER"),
+		os.Getenv("ONEVIEW_OV_PASSWORD"),
+		os.Getenv("ONEVIEW_OV_DOMAIN"),
+		os.Getenv("ONEVIEW_OV_ENDPOINT"),
+		false,
+		600)
+
+	interconnect_list, err := ovc.GetInterconnects("", "", "", "")
```
Comment (soodpr) [inline examples/interconnect.go]: you can add some examples for getting interconnects using filter or other params.

Comment (jyothis-gireesan) [reply examples/interconnect.go]: please check now.

File: ov/interconnect.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+	EnableStormControl            bool                 `json:"enableStormControl,omitempty"`            // "enableStormControl": false,
+	EnableTaggedLldp              bool                 `json:"enableTaggedLldp,omitempty"`              // "enableTaggedLldp": false,
+	EnclosureName                 string               `json:"enclosureName,omitempty"`                 // "enclosureName": "SYN03_Frame1",
+	EnclosureType                 string               `json:"enclosureType,omitempty"`                 // "enclosureType": "SY12000",
+	EnclosureUri                  utils.Nstring        `json:"enclosureUri,omitempty"`                  // "enclosureUri": "/rest/enclosures/013645CN759000AC",
+	ETag                          string               `json:"eTag,omitempty"`                          // "eTag": "463bd328-ffc8-40ae-9603-6136fa9e6e58",
+	FirmwareVersion               string               `json:"firmwareVersion,omitempty"`               // "firmwareVersion": "1.3.0.1005",
+	HostName                      string               `json:"hostName,omitempty"`                      // "hostName": "VC4040F8-2TV5451754",
```
Comment (soodpr) [inline ov/interconnect.go]: @jyothis-gireesan - can you please check if we have all the attributes like icmLicenses, initialScopeUris ar missing here

**Comment by soodpr** [comment]
@jyothis-gireesan - can you please rebase it with master as png files have been added in the [gitattributes](https://github.com/HewlettPackard/oneview-golang/blob/master/.gitattributes) file in master

File: endpoints-support.md [hunk 3]
```diff
@@ -52,6 +52,9 @@
 |<sub>/rest/fcoe-networks/{id}</sub>                                                      | PATCH    | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_multiplication_x:   |
 |<sub>/rest/fcoe-networks/{id}</sub>                                                      | PUT      | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   |
 |<sub>/rest/fcoe-networks/{id}</sub>                                                      | DELETE   | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   |
+|     **Interconnect**                                                                                                                          |
```
Comment (madhav-bharadwaj) [inline endpoints-support.md]: resource name can be changed to `interconnects`

File: examples/interconnect.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+		// new_eg_name = "RenamedEnclosureGroup"
+		// script      = "#TEST COMMAND"
+	)
+	ovc := clientOV.NewOVClient(
+		os.Getenv("ONEVIEW_OV_USER"),
+		os.Getenv("ONEVIEW_OV_PASSWORD"),
+		os.Getenv("ONEVIEW_OV_DOMAIN"),
+		os.Getenv("ONEVIEW_OV_ENDPOINT"),
+		false,
+		800)
```
Comment (dileepds) [inline examples/interconnect.go]: Need one more parameter ifmatch for NewOvClient

**Pull request merged**